Userland/Libraries/LibAudio/FlacLoader.cpp

```diff
@@ -145,6 +148,38 @@ MaybeLoaderError FlacLoaderPlugin::parse_header()
     return {};
 }
 
+// 11.19. METADATA_BLOCK_PICTURE
+MaybeLoaderError FlacLoaderPlugin::load_picture(FlacRawMetadataBlock& block)
+{
+    auto memory_stream = LOADER_TRY(Core::Stream::MemoryStream::construct(block.data.bytes()));
+    auto picture_block_bytes = LOADER_TRY(BigEndianInputBitStream::construct(*memory_stream));
+
+    PictureData picture {};
+
+    picture.type = static_cast<FlacPictureType>(LOADER_TRY(picture_block_bytes->read_bits(32)));
+
+    auto const mime_string_length = LOADER_TRY(picture_block_bytes->read_bits(32));
+    picture.mime_string = { block.data.bytes().data() + memory_stream->offset(), (size_t)mime_string_length };
```

Comment:
This will create a string from out of bounds data if the length is set maliciously large; in general see <https://datatracker.ietf.org/doc/html/draft-ietf-cellar-flac-05#section-13>. Please check the length first.

Reply:
Sure, I feel so naive right know haha.
How do you want me to check this ?
Trusting the input while the total read size is still inside the block size?

I just had a quick glance, I will answer all of your comments later.